### PR TITLE
AP_HAL_ChibiOS: use structure for writing out WDOG,MON message

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
@peterbarker, I haven't checked this in detail yet but I'm wondering why we are changing this?  I guess for performance reasons?

**Comment by andyp1per** [comment]
@rmackay9 its because otherwise we are allocating FMT messages in the watchdog thread when things blow up giving the possibility of even worse blow up - so its a safety fix.

**Review by andyp1per — APPROVED** [review]

**Comment by peterbarker** [comment]
> @peterbarker, I haven't checked this in detail yet but I'm wondering why we are changing this? I guess for performance reasons?

This came up as part of my discussions with Andy the other day, and again today with our deep-delve into our all-zeroes watchdog resets.

`AP::logger().LogWrite(...)`, while really quite convenient in terms of being able to drop all the information about a message in one place, is very much not the most efficient way of doing things.

One way it it not efficient is that the first time we emit the message we have to go and allocate structures, do a bunch of memcpy'ing etc etc, to add a persistently known format into AP_Logger's structures.

Crossing that codepath increased maximum stack usage on the monitor stack to the point we got a stack overflow.  So in the middle of writing out a `MON` message - so the system is already in trouble! - we go off and overwrite the stack.

So this PR stops using anywhere near as much stack in the monitor thread when we do actually trigger a long loop.

We believe this can also trigger a continuous watchdog reset loop.

Basically, if you're in a hole, stop digging....

File: libraries/AP_HAL_ChibiOS/Scheduler.cpp [hunk 1]
```diff
@@ -410,19 +410,23 @@ void Scheduler::_monitor_thread(void *arg)
...
-                                   pd.internal_errors,
-                                   pd.internal_error_count,
-                                   pd.internal_error_last_line,
-                                   pd.last_mavlink_msgid,
-                                   pd.last_mavlink_cmd,
-                                   pd.semaphore_line,
-                                   pd.spi_count,
-                                   pd.i2c_count);
-                }
+                gcs().send_text(MAV_SEVERITY_INFO, "writing MON packet");
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/Scheduler.cpp:413]: debug left over

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/Scheduler.cpp]: Fixed.
